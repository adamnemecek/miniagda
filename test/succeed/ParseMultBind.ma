let K : (A, B : Set) -> Set
    = \ A B -> A

data Prod ++(A, B : Set) : Set
{ pair : A -> B -> Prod A B
}

fun fst : [A, B : Set] -> Prod A B -> A
{ fst A B (pair a b) = a
}
