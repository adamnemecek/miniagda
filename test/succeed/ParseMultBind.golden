MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "ParseMultBind.ma" ---
--- scope checking ---
--- type checking ---
type  K : (A : Set) -> (B : Set) -> Set
type  K = \ A -> \ B -> A
type  Prod : ++(A : Set) -> ++(B : Set) -> Set
term  Prod.pair : .[A : Set] -> .[B : Set] -> ^(y0 : A) -> ^(y1 : B) -> < Prod.pair y0 y1 : Prod A B >
term  fst : .[A : Set] -> .[B : Set] -> Prod A B -> A
{ fst [A] [B] (Prod.pair a b) = a
}
term  snd : .[A : Set] -> .[B : Set] -> (p : Prod A B) -> B
{ snd [A] [B] (Prod.pair a b) = b
}
--- evaluating ---
--- closing "ParseMultBind.ma" ---
